client/src/components/LeafletOsmMap/index.js

```diff
@@ -171,6 +172,13 @@ export default class LeafletOSMMap extends Component {
       autoPan: true
     }).addTo(this.state.map);
 
+    if (this.state.unSavedMarker) {
+          console.log(this.state.unSavedMarker);
+          this.state.unSavedMarker.remove();
+          dropPin(this.state.UnSavedMarker, event.target);
```

Comment:
I don't think you need this code snippet - this function is supposed to assist add a pin to the map. Isn't the purpose of this if scope to remove any "unSavedMarker" from the map?